src/allmydata/scripts/tahoe_add_alias.py

```diff
@@ -111,34 +151,42 @@ def _get_alias_details(nodedir):
     return data
 
 
-def list_aliases(options):
-    nodedir = options['node-directory']
-    stdout = options.stdout
-    stderr = options.stderr
+def _escape_format(t):
+    """
+    _escape_format(t).format() == t
 
-    data = _get_alias_details(nodedir)
+    :param unicode t: The text to escape.
+    """
+    return t.replace("{", "{{").replace("}", "}}")
 
-    max_width = max([len(quote_output(name)) for name in data.keys()] + [0])
-    fmt = "%" + str(max_width) + "s: %s"
-    rc = 0
+
+def list_aliases(options):
+    """
+    Show aliases that exist.
+    """
+    data = _get_alias_details(options['node-directory'])
 
     if options['json']:
-        try:
-            # XXX why are we presuming utf-8 output?
-            print(json.dumps(data, indent=4).decode('utf-8'), file=stdout)
-        except (UnicodeEncodeError, UnicodeDecodeError):
-            print(json.dumps(data, indent=4), file=stderr)
-            rc = 1
+        output = _escape_format(json.dumps(data, indent=4).decode("utf-8"))
```

Comment:
Minor comment.

I am not sure why `decode utf-8` is needed here.
I guess that this is just a conversion from bytes to unicode... decode('ascii') will also work.

I prefer to see .decode('ascii') as it means that the source is not encoded in utf-8

or

```suggestion
        output = _escape_format(json.dumps(data, indent=4, ensure_ascii=False))
```

Reply:
Also, I guess that you can just show the output and return as json dump is never an empty string... this will remove the usage of `else`

```
show_output(options.stdout, output)
return 0
```

Reply:
Ah yea, ASCII is a better codec choice here and `ensure_ascii` looks like a reasonable way to get there.  I didn't previously realize `ensure_ascii` was a way of controlling the result type, I thought it was about the characters that end up in the result value.

Reply:
Although even after seeing the behavior and reading the docs repeatedly, the docs still don't make any sense to me.  I might just go with `.decode("ascii")` because at least I understand what it means.